flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -519,6 +519,12 @@ object JobManager {
           configuration.setString(ConfigConstants.FLINK_BASE_DIR_PATH_KEY, config.configDir + "/..")
         }
 
+        if (GlobalConfiguration.getString(ConfigConstants.JOB_MANAGER_IPC_ADDRESS_KEY,
+          null) == null) {
+          configuration.setString(ConfigConstants.JOB_MANAGER_IPC_ADDRESS_KEY,
+            InetAddress.getLocalHost.getHostName)
+        }
+
         val hostname = configuration.getString(ConfigConstants.JOB_MANAGER_IPC_ADDRESS_KEY, null)
```

Comment:
In local mode, jobmanager will start an internal taskmanager and  call the `TaskManager.parseConfiguration`. For TaskManager, we can replace the null default value with the `config.defaultJobManagerAdd` but not the `InetAddress.getLocalHost.getHostName`. I think the question is whether we add an additional parameter to the `parseConfiguration` for the default jobmanager address or put this in the configuration.